builder/amazon/ebsvolume/step_tag_ebs_volumes.go

```diff
@@ -36,34 +38,85 @@ func (s *stepTagEBSVolumes) Run(ctx context.Context, state multistep.StateBag) m
+		// volume will have had these tags applied when the instance was
+		// created. We now need to remove these tags to ensure only the EBS
+		// volume tags are applied (if any)
+		toDeTag := map[string][]*ec2.Tag{}
+		if config.VolumeRunTags.IsSet() {
+			ui.Say("Removing any tags applied to EBS volumes when the source instance was created...")
 
+			ui.Message("Compiling list of existing tags to remove...")
+			existingTags, err := config.VolumeRunTags.EC2Tags(s.Ctx, *ec2conn.Config.Region, state)
+			if err != nil {
+				err := fmt.Errorf("Error generating list of tags to remove: %s", err)
+				state.Put("error", err)
+				ui.Error(err.Error())
+				return multistep.ActionHalt
+			}
+			existingTags.Report(ui)
+
+			// Generate the map of volumes and associated tags to delete.
+			// Looping over the instance block device mappings allows us to
+			// obtain the volumeId
+			for _, mapping := range s.VolumeMapping {
+				for _, v := range instance.BlockDeviceMappings {
+					if *v.DeviceName == mapping.DeviceName {
+						toDeTag[*v.Ebs.VolumeId] = existingTags
+					}
+				}
+			}
+
+			// Delete the tags
+			for volumeId, existingTags := range toDeTag {
+				ui.Message(fmt.Sprintf("Deleting existing 'run_volume_tags' on EBS Volume: %s", volumeId))
+				_, err = ec2conn.DeleteTags(&ec2.DeleteTagsInput{
+					Resources: aws.StringSlice([]string{volumeId}),
+					Tags:      existingTags,
+				})
+				if err != nil {
+					err := fmt.Errorf("Error deleting existing tags on EBS Volume %s: %s", volumeId, err)
+					state.Put("error", err)
+					ui.Error(err.Error())
+					return multistep.ActionHalt
+				}
+			}
+		}
+
+		ui.Say("Tagging EBS volumes...")
 		toTag := map[string][]*ec2.Tag{}
 		for _, mapping := range s.VolumeMapping {
+
 			if len(mapping.Tags) == 0 {
 				ui.Say(fmt.Sprintf("No tags specified for volume on %s...", mapping.DeviceName))
 				continue
 			}
 
+			ui.Message("Compiling list of tags to apply to volume...")
 			tags, err := mapping.Tags.EC2Tags(s.Ctx, *ec2conn.Config.Region, state)
 			if err != nil {
-				err := fmt.Errorf("Error tagging device %s with %s", mapping.DeviceName, err)
+				err := fmt.Errorf("Error generating tags for device %s: %s", mapping.DeviceName, err)
 				state.Put("error", err)
 				ui.Error(err.Error())
 				return multistep.ActionHalt
 			}
 			tags.Report(ui)
 
+			// Generate the map of volumes and associated tags to apply.
+			// Looping over the instance block device mappings allows us to
+			// obtain the volumeId
 			for _, v := range instance.BlockDeviceMappings {
 				if *v.DeviceName == mapping.DeviceName {
 					toTag[*v.Ebs.VolumeId] = tags
 				}
 			}
 		}
 
+		// Apply the tags
 		for volumeId, tags := range toTag {
+			ui.Message(fmt.Sprintf("Applying tags to EBS Volume: %s", volumeId))
 			_, err := ec2conn.CreateTags(&ec2.CreateTagsInput{
-				Resources: []*string{&volumeId},
+				Resources: aws.StringSlice([]string{volumeId}),
```

Comment:
same here, can we apply tags to multiple volumes at once ?

Reply:
Not here unfortunately. The tags for each volume in `ebs_volumes` are configured separately - in other words each volume has its own individual set of tags. As such, we need to make an API call per volume.

Reply:
gotcha ! 👍